### ISSUE TITLE
UI for mass acknowledge/maintenance/delete for checks

### ISSUE THREAD

**Issue by necula01** [flapjack/flapjack#873]
We have a large number of entities and checks and it would be very useful to be able to issue mass operations on checks. 

The way I envision it is to show a checkbox next to each check in the various tables showing checks (Checks, Failing Checks, Entity). 
* Then the UI would allow various ways to select a number of these checkboxes
  * drill-down with the table sorter filters, 
  * select All/None that are displayed, 
  * shift and ctrl-click to select/deselect ranges of contiguous boxes. 
* Finally, there will be buttons for Acknowledge, Schedule Maintenance, End Maintenance, Decommission.   
  * upon pressing one of these buttons, a form with be shown at the top to enter the summary and duration, as necessary, along with a Cancel button that hides the form.
* Upon pressing Ok, a POST message will be sent to the server. The pure REST design would suggest multiple POST messages, but I think given how the web.rb is implemented a single POST should be sent, with the url, e.g., /scheduled_maintenances/_many and the ids of the entities and checks encoded in the payload. 

Is anybody working on this? Is the workflow acceptable? Do you want to suggest refinements?

**Comment by ali-graham** [comment]
Hi George, I'm currently working (in a branch off of `master`, which is all quite different under the hood) to make the `web` pikelet a client of Flapjack's API. The API for the next major release is implemented to be compliant with v1 of the [JSONAPI spec](http://jsonapi.org) and is (not yet fully) documented using [Swagger](http://swagger.io). 

We are hoping to have a pre-release of Flapjack 2.0 out soon -- it would probably be much easier to construct the interface you describe using that, as things like bulk operations, object relationships etc. all operate according to externally specified and documented behaviour, and any API operations you write would be going through [flapjack-diner](https://github.com/flapjack/flapjack-diner/).

The workflow sounds useful -- I already have [refreshable filters](https://github.com/flapjack/flapjack/blob/10014d9d0aa52ef1ee896dee92a454d328bb5446/lib/flapjack/gateways/api_web/views/checks.html.erb#L13-L55) working in that branch, but haven't yet caught up on the various POST/PATCH operations on checks. We'll be aiming to release something that at least has parity with v1's web interface, and can then improve going forward.

tl;dr -- it'd be a fair bit of effort in v1, and a lot of the work to make things easier has been done for v2.

**Comment by necula01** [comment]
That sounds like a good plan, but I have not been able to find much documentation for v2. Do you have an idea of when I could play with a pre-release version?

**Comment by ali-graham** [comment]
Documentation is pretty much the last thing on the list, unfortunately :( We're aiming to at least get something out for people to poke at before [Monitorama](http://monitorama.com/), so roughly 2 weeks, but that may depend on at least getting a minimum level of useful docs ready as well.

**Comment by necula01** [comment]
I want to restart the discussion of mass acknowledgements in the UI. We have 100k checks on 1000 entities and we need to do mass operations. I have an implementation that is working, I would like to merge it in your branch so I can more easily keep in sync.
